hibernate-core/src/main/java/org/hibernate/sql/ast/tree/from/UnionTableGroup.java

```diff
@@ -63,24 +65,37 @@ public LockMode getLockMode() {
 
 	@Override
 	public Set<TableGroupJoin> getTableGroupJoins() {
-		return Collections.emptySet();
+		return tableGroupJoins == null ? Collections.emptySet() : Collections.unmodifiableSet( tableGroupJoins );
 	}
 
 	@Override
 	public boolean hasTableGroupJoins() {
-		return false;
+		return tableGroupJoins != null && !tableGroupJoins.isEmpty();
 	}
 
 	@Override
 	public void setTableGroupJoins(Set<TableGroupJoin> joins) {
+		if ( tableGroupJoins == null ) {
+			tableGroupJoins = new HashSet<>( joins );
+		}
+		else {
+			tableGroupJoins.addAll( joins );
+		}
```

Comment:
seems a little bit confusing for what is done is `addAllTableGroupJoins()`, not replacing the previous content.

Reply:
Fixed, thanks Nathan